### PR TITLE
#206 add close method to OracleClient

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java [hunk 1]
```diff
@@ -277,8 +316,8 @@ public OracleClient(Environment env) {
     responseTimer = env.getSharedResources().getMetricRegistry().timer(env.getMeticNames().getOracleClientGetStamps());
     stampsHistogram = env.getSharedResources().getMetricRegistry().histogram(env.getMeticNames().getOrcaleClientStamps());
 
-    // TODO make thread exit if idle for a bit, and start one when request arrives
-    Thread thread = new Thread(new TimestampRetriever());
+    timestampRetriever = new TimestampRetriever();
+    thread = new Thread(new TimestampRetriever());
```
Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: should this be `thread = new Thread(timestampRetriever);` ?

Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: yes it should... nice catch

Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: getting lots of exception when running ITs after making that change

Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: I fixed this e9eb07f .   The test are running, but I Am seeing more exceptions now.  I will make a comment about the line thats causing this.  Not sure what we can do about these extra exceptions.

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java [hunk 2]
```diff
@@ -212,13 +230,28 @@ private synchronized void connect() throws IOException, KeeperException, Interru
...
     }
 
-    private void close() {
-      if(transport.isOpen())
+    private synchronized void close() {
+      if(transport != null && transport.isOpen())
         transport.close();
+      try {
+        if(pathChildrenCache != null)
+          pathChildrenCache.close();
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java:243]: Closing this seems to cause a background thread created by this object to log execeptions.  This is unfortunate, not sure what we can do other than log a curator bug.   It does not cause test to fail, just causes noise.

Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java:243]: After we get this merged in, I can look into opening a Fluo bug and upstream curator bug.

**Comment by mikewalch** [comment]
+1.. can you squash your commits into one?

**Comment by keith-turner** [comment]
squashed

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java (hunk 2 above)
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java:243]: I opened fluo-io/fluo#418

**Pull request merged**